haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs

```diff
@@ -177,82 +194,72 @@ declsFromRenamed ctx (hsGroup, _, _, _) =
-                --
-                -- "Qualified currently behaves exactly as Implicit, but it is
-                -- deprecated to use it for implicit quantification."
+    mkDeclName n = nameDeclAlt ctx n Nothing typeStringyType
+    explicitNames = concatMap namesFromForall . universeBi $ hsGroup
+    implicitVarBindDefs =
+      let allUsages = concatMap namesFromTyVar . universeBi $ hsGroup
+          firstUsages = mapMaybe keepFirst
+            . groupBy ((==) `on` fst)
+            . sortBy (comparing fst)
+            . map fromLoc
+            $ allUsages
+      in map mkDecl firstUsages
       where
-        implicitCase =
-            let boundTyVarNames = Set.fromList
-                                . map (boundTyVarName . unLoc) . hsq_tvs
-                                $ binders
-                usages = mapMaybe (interestingNameToLoc boundTyVarNames)
-                       . universeBi
-                       $ (ty, context)
-                firstUsages = mapMaybe keepFirst
-                            . groupBy ((==) `on` fst)
-                            . sortBy (comparing fst)
-                            $ usages
-            in map mkDecl firstUsages
-                 where mkDecl (n,l) = declWithWrappedIdLoc typeStringyType
-                                                           (L l n)
-        --
-        interestingNameToLoc s (L loc (HsTyVar n))
-            | Set.member n s = Just $! (n, loc)
-            | otherwise = Nothing
-        interestingNameToLoc _ _ = Nothing
+        mkDecl (n,l) = declWithWrappedIdLoc typeStringyType (L l n)
+        fromLoc (L l n) = (n, l)
         keepFirst [] = Nothing
         keepFirst xs@((name, _):_) = Just $! (name, minimum $ map snd xs)
-    declsFromForall _ = []
+
+    namesFromTyVar :: HsType Name -> [Located Name]
+    namesFromTyVar (HsTyVar n) = [n]
+    namesFromTyVar _ = []
+    namesFromForall :: HsType Name -> [Name]
+    namesFromForall (HsForAllTy binders _) = map hsLTyVarName binders
+      where
+        mkDecl binder =
+          nameDeclAlt ctx (boundTyVarName binder) Nothing typeStringyType
+    namesFromForall _ = []
     -- | For typevar binders directly attached to datatype declarations (like
     -- data, type, class etc), the location of the name is correctly set to
     -- the span of the type variable.
     -- TODO(robinpalotai): as in declsFromForall about stripping kind signature
     --   from span.
+#if __GLASGOW_HASKELL__ >= 800
+    declsFromDataBinders :: LHsQTyVars Name -> [DeclAndAlt]
+#else
     declsFromDataBinders :: LHsTyVarBndrs Name -> [DeclAndAlt]
-    declsFromDataBinders = map (mkDecl . boundTyVarName . unLoc) . hsq_tvs
+#endif
+    --TODO: is this missing something
+    declsFromDataBinders = map (mkDecl . boundTyVarName . unLoc) . hsq_explicit
       where
         mkDecl n = nameDeclAlt ctx n Nothing typeStringyType
+    -- This function exists and is called hsTyVarName in GHC 8.0
```

Comment:
either prefix with TODO, or feel free to make `boundTyVarName = hsTyVarName`, or even rename the old one (and it will only be defined in GHC 7).